### PR TITLE
feature #65984-Variant Color Configuration Managing

### PR DESCRIPTION
**Pull request by Anstsiya**

### CODE REVIEW

**Review by InnaBoitsun — COMMENTED** [review]
I did not check yet the implementation but I see next visual issues:
1) it is not possible to close Color Picker when it is opened - clicking outside did not help
2) when filter is used it shows old colors in Picker (from prev items) 
image
3) is it possible to show white instead of black when color is absent?
4) I was not able to Save - I see error in JS:
image
5) when all colors are rendered it looks like C1 hangs on and it is not possible to switch tabs
- need to think more here to have better implementation ( some paging or  show first XXX items and "show more" btn at bottom)

**Comment by Anstsiya** [comment]
resolved comments for now except 
5. when all colors are rendered it looks like C1 hangs on and it is not possible to switch tabs
need to think more here to have better implementation ( some paging or show first XXX items and "show more" btn at bottom)

**Review by InnaBoitsun — COMMENTED** [review]
Color Pciker doesn work, It does not display selected color and when click on input, the selected color is not applied

File: src/Orckestra.Composer.Product/GraphQL/Extensions/DataGraphQLExtensions.cs [hunk 1]
```diff
@@ -0,0 +1,105 @@
+﻿using Composite.Data;
+using GraphQL;
+using GraphQL.Types;
+using System;
+using System.Linq;
+using Orckestra.Composer.Product.GraphQL.Types;
+
+namespace Orckestra.Composer.Product.GraphQL.Extensions
+{
+    public static class DataGraphQLExtensions
```
Comment (neexite) [inline src/Orckestra.Composer.Product/GraphQL/Extensions/DataGraphQLExtensions.cs]: Move reusable classes to the common dll (Composer or separated library)

File: src/Orckestra.Composer.Product/GraphQL/Extensions/LambdaExtensions.cs [hunk 2]
```diff
@@ -0,0 +1,22 @@
+﻿using System;
+using System.Linq.Expressions;
+using System.Reflection;
+
+namespace Orckestra.Composer.Product.GraphQL.Extensions
```
Comment (neexite) [inline src/Orckestra.Composer.Product/GraphQL/Extensions/LambdaExtensions.cs]: Move reusable classes to the common dll (Composer or separated library)

File: src/Orckestra.Composer.Product/GraphQL/Extensions/LambdaExtensions.cs [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+{
+    public static class LambdaExtensions
```
Comment (neexite) [inline src/Orckestra.Composer.Product/GraphQL/Extensions/LambdaExtensions.cs]: to common dll

File: src/Orckestra.Composer.Product/GraphQL/Types/DataInput.cs [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Composite.Data;
+using Orckestra.Overture.ServiceModel;
+
+namespace Orckestra.Composer.Product.GraphQL.Types
+{
+    public class DataInput<TSourceType> where TSourceType : class, IData
```
Comment (neexite) [inline src/Orckestra.Composer.Product/GraphQL/Types/DataInput.cs]: to common dll

File: src/Orckestra.Composer.Product/GraphQL/Types/DataInputObjectGraphType.cs [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+using System.Collections.Generic;
+using System.Linq;
+using System.Linq.Expressions;
+using System.Reflection;
+using System.Runtime.ExceptionServices;
+using Orckestra.Composer.Product.GraphQL.Extensions;
+
+namespace Orckestra.Composer.Product.GraphQL.Types
+{
+    public class DataInputObjectGraphType<TSourceType> : AutoRegisteringInputObjectGraphType<TSourceType> where TSourceType : class, IData
```
Comment (neexite) [inline src/Orckestra.Composer.Product/GraphQL/Types/DataInputObjectGraphType.cs]: to common dll

File: src/Orckestra.Composer.Product/GraphQL/Types/DataObjectGraphType.cs [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+using GraphQL;
+using GraphQL.Types;
+using System;
+using System.Linq;
+using System.Linq.Expressions;
+using Orckestra.Composer.Product.GraphQL.Extensions;
+
+namespace Orckestra.Composer.Product.GraphQL.Types
+{
+    public class DataObjectGraphType<TSourceType> : AutoRegisteringObjectGraphType<TSourceType>
```
Comment (neexite) [inline src/Orckestra.Composer.Product/GraphQL/Types/DataObjectGraphType.cs]: to common dll

**Review by InnaBoitsun — COMMENTED** [review]
After the Save the new colors disapper from inputs - see creen:
KTjD5fcF85

File: src/Orckestra.Composer.Product/StartupHandler.cs [hunk 7]
```diff
@@ -0,0 +1,94 @@
...
+
+
+        public static void OnBeforeInitialize()
+        {
+
+        }
+
+        public static void OnInitialized()
+        {
+            EnsureCreateStore();
```
Comment (neexite) [inline src/Orckestra.Composer.Product/StartupHandler.cs]: Is it required?

File: tests/Orckestra.Composer.Sitemap.Tests/app.config [hunk 8]
```diff
@@ -26,6 +26,26 @@
         <assemblyIdentity name="Moq" publicKeyToken="69f491c39445e920" culture="neutral" />
         <bindingRedirect oldVersion="0.0.0.0-4.2.1507.118" newVersion="4.2.1507.118" />
       </dependentAssembly>
+      <dependentAssembly>
+        <assemblyIdentity name="System.Web.Razor" publicKeyToken="31bf3856ad364e35" culture="neutral" />
+        <bindingRedirect oldVersion="0.0.0.0-3.0.0.0" newVersion="3.0.0.0" />
```
Comment (neexite) [inline tests/Orckestra.Composer.Sitemap.Tests/app.config]: why it there?

File: src/Orckestra.Composer.Product/StartupHandler.cs [hunk 9]
```diff
@@ -0,0 +1,94 @@
...
+using Orckestra.Composer.Services.Lookup;
+
+namespace Orckestra.Composer.Product
+{
+    [ApplicationStartup(AbortStartupOnException = true)]
+    public static class StartupHandler
+    {
+        public static void ConfigureServices(IServiceCollection collection)
+        {
+            //GraphQL, NOTE: may be move to IRecipeDocumentExecuter and IRecipeDocumentWriter
```
Comment (neexite) [inline src/Orckestra.Composer.Product/StartupHandler.cs]: Implement notes :-)
Think about reusing

**Review by neexite — CHANGES_REQUESTED** [review]

**Review by neexite — APPROVED** [review]

**Pull request merged**